docs/06-using-cairo.md

```diff
@@ -0,0 +1,160 @@
+# Using Cairo and LGI
+
+These days, Awesome interface is mostly based on a library called LGI. It allows
+to access C libraries such as GTK, GLib, Cairo, Pango, PangoCairo and RSVG using
+Lua code without having to write actual "glue" C code.
+
+This is done using the GObject-introspection framework. The main advantage is
+the time saved and large number of features exposed for free. The downside is
+the lack of proper Lua contric documentation and examples. Some examples can be
+found in [LGI own documentation](https://github.com/pavouk/lgi/tree/master/docs)
+, but this doesn't even cover 1% of the actual API. Using other APIs require
+some trial and errors. It can also prove impossible if the introspection data
+is missing in inacurate. Using those low level APIs directly is also crash
+prone. It is the programmer responsability to properly check return and errors
+values.
+
+## Using LGI in Awesome
+
+GObject and Gnome centric libraries tend to use the common C practive of
+emulating namespaces using underscores in funciton name. LGI expose a proper
+namespace based API. For example, if the C function is:
+
+    cairo_image_surface_create()
+
+Then LGI equivalent is:
+
+    lgi.cairo.ImageSurface.create()
+
+The same goes for enums:
+
+    CAIRO_FORMAT_ARGB32
+
+become:
+
+    lgi.cairo.Format.ARGB32
+
+LGI is also object oriented while the C API is function based. When those
+function take the "class" "object", then this:
+
+    cairo_line_to(cr, x, y)
+
+can be expressed as:
+
+    cr:line_to(x, y)
+
+It is however important to note some inconsistencies. For example, Cairo is
+called `lgi.cairo` while GLib is called `lgi.GLib`. Figuring this out will
+require some tried and errors. The best way to do this without actually
+reloading Awesome is to open the `lua` command in a terminal and using `print`:
+
+    print("This will print a table address:", require("lgi").cairo)
+    print("This will print an error:", require("lgi").Cairo)
+
+It is recommanded to avoid using `require` everytime you use a function and
+instead include the libraries at the top of your `rc.lua` or lua module
+
+    local cairo = require("lgi").cairo
+
+## The Cairo API
+
+Cairo is a 2D graphic library used by Awesome, Gnome and XFCE. It allows to
+paint paths on a `surface`. Awesome use it internally and being able to call
+is directly should be considered a power user feature.
+
+Understanding these few important concepts is necessary to be able to use Cairo:
+
+**Surface**:
+
+A surface is the area where the painting will be done. There is multiple types
+of surfaces including:
+
+ * Color pixmap with transparency (`ARGB32`) or without (`RGB24`)
```

Comment:
Can we call these images instead of pixmaps? That's what cairo actually calls them. Also, pixmap in an X11-context has a very specific meaning that is different from this one.
(Same for the following lines)